tests/std/include/test_header_units_and_modules.hpp

```diff
@@ -0,0 +1,1088 @@
+        334, 167, 502, 251, 754, 377, 1132, 566, 283, 850, 425, 1276, 638, 319, 958, 479, 1438, 719, 2158, 1079, 3238,
+        1619, 4858, 2429, 7288, 3644, 1822, 911, 2734, 1367, 4102, 2051, 6154, 3077, 9232, 4616, 2308, 1154, 577, 1732,
+        866, 433, 1300, 650, 325, 976, 488, 244, 122, 61, 184, 92, 46, 23, 70, 35, 106, 53, 160, 80, 40, 20, 10, 5, 16,
+        8, 4, 2, 1, -1000};
+    assert(equal(vec.begin(), vec.end(), begin(expected_val), end(expected_val)));
+}
+
+void test_streambuf() {
+    using namespace std;
+    puts("Testing <streambuf>.");
+    istringstream iss{"kittens"};
+    assert(iss.rdbuf()->in_avail() == 7);
+}
+
+void test_string() {
+    using namespace std;
+    puts("Testing <string>.");
+    const string small_string{"homeowner"};
+    const string large_string{"Cute fluffy kittens are so adorable when they meow and purr."};
+    assert(small_string.find("meow") == 2);
+    assert(large_string.find("meow") == 46);
+}
+
+void test_string_view() {
+    using namespace std;
+    puts("Testing <string_view>.");
+    constexpr string_view catenary{"catenary"};
+    assert(catenary.starts_with("cat"));
+    assert(!catenary.starts_with("dog"));
+    static_assert(catenary.starts_with("cat"));
+    static_assert(!catenary.starts_with("dog"));
+}
+
+#pragma warning(push)
+#pragma warning(disable : 4996) // was declared deprecated
+void test_strstream() {
+    using namespace std;
+    puts("Testing <strstream>.");
+    istrstream istr{"1729"};
+    int n = -1;
+    istr >> n;
+    assert(n == 1729);
+}
+#pragma warning(pop)
+
+void test_syncstream() {
+    using namespace std;
+    puts("Testing <syncstream>.");
+    syncbuf sync_buf{nullptr};
+    assert(sync_buf.get_wrapped() == nullptr);
+    assert(sync_buf.get_allocator() == allocator<char>{});
+    assert(sync_buf.emit() == false);
+    osyncstream sync_str{cout};
+    sync_str << "Testing osyncstream.\n";
+    assert(sync_str.rdbuf()->get_wrapped() == cout.rdbuf());
+}
+
+void test_system_error() {
+    using namespace std;
+    puts("Testing <system_error>.");
+    const error_code code = make_error_code(errc::value_too_large);
+    assert(code.value() == static_cast<int>(errc::value_too_large));
+    assert(code.category() == generic_category());
+    assert(code);
+}
+
+void test_thread() {
+    using namespace std;
+    puts("Testing <thread>.");
+    // see above, tested with <condition_variable>
+    assert(this_thread::get_id() != thread::id{});
+}
+
+void test_tuple() {
+    using namespace std;
+    puts("Testing <tuple>.");
+    constexpr tuple<int, char, double> t{1729, 'c', 1.25};
+    assert(get<int>(t) == 1729);
+    assert(get<char>(t) == 'c');
+    assert(get<double>(t) == 1.25);
+    static_assert(get<int>(t) == 1729);
+    static_assert(get<char>(t) == 'c');
+    static_assert(get<double>(t) == 1.25);
+}
+
+void test_type_traits() {

```

Comment:
Grumble, grumble "`<typeindex>` and `<typeinfo>` precede `<type_traits>` in tab:headers.cpp".